### ISSUE TITLE
Cannot add a virtual keyboard on Bazcor 1.4.0

### ISSUE THREAD

**Issue by lrvega** [Dygmalab/Bazecor#828]
**Describe the bug**
Unable to add a virtual keyboard on [Bazcor 1.4.0](https://github.com/Dygmalab/Bazecor/releases/tag/v1.4.0) running on Linux Mint 20.3 Cinnamon

**To Reproduce**
1. Start Bazcor either by double clicking on the executable or running from the command line
2. Click on "Add Virtual Keyboard" either from the top button next to "Scan Keyboards" or from the "Add keyboard" section
3. Pick a model from the "Select model" pull down menu (any model will do, I tried them all an none worked)
4. Click on create new file
5. Save the file
6. Saved file is not created, hence no virtual keyboard is displayed

**Expected behavior**
That the virtual keyboard is created and available for configuration

**Desktop (please complete the following information):**
 - OS: Linux Mint 20.3 Cinnamon
 - Bazecor Version: 1.4.0

**Additional context**
I ran the executable from the command line and didn't see any logs that would help track the error down:
```
08:17:44.338 › Devices Available: []
08:17:44.339 › EXECUTED addDevicesList:  [] []
08:17:44.339 › Current State:  [] 0
08:17:45.864 › Add virtual device!
08:17:45.865 › Current State:  [] 0
```

**Comment by lrvega** [comment]
I have also tried this in a Ubuntu 24.04 VM and the results are the same.

**Comment by jolitp** [comment]
I encountered the same issue. The last version that allows adding a virtual keyboard (from the ones I tried) is v1.3.11.

**Comment by noticons** [comment]
I am running openSUSE Tumbleweed, latest version, and Bazcore 1.4.1.
I'm getting the same problem. When I run the appimage via the terminal, there is no output after asking it to save the json file. Then, if I go into Preferences, it hangs and doesn't go anywhere. The Loading Bazcore icon, however, does animate, and the terminal shows that some action has occured. If I shut off Bazcore, it is still running, since the terminal hasn't escaped. And a look at the system monitor shows, of course, that it is still running. Bazcore has done all the proper steps for unloading however, according to the terminal. The last step reading: "10:47:18.952 › removing USB Listeners" 
This is just my guess, but I'm guessing that Bazcore hangs because some error message is trying (but failing) to be produced concerning a failure to create the file--for whatever reason.
If, for instance, I restart the app via the terminal, and don't try to create a virtual keyboard, I can do everything, including exit Bazcore, and everything works perfectly, including a clean terminal escape on closing.

**Comment by alexpargon** [comment]
Hey there! is this a permissions thing on the destination folder? or something related with APP permissions as a whole... 

can you try it out to a folder with `chmod 777`? just to be sure it's not a weird interaction with the permissions system for whatever odd reason (like a dependency update that changed the way this works)

**Comment by lrvega** [comment]
For what it's worth, I don't think it's a permission issue. Bazecor 1.3.x can create virtual keyboards.
```
🐧 ❯ ls -l | grep Dygma
drwxrwxr-x  3 lrvega lrvega   4096 Jul 13 08:16 Dygma
~ 
🐧 ❯ cd Dygma
~/Dygma 
🐧 ❯ ls -l 
total 4
drwxrwxr-x 2 lrvega lrvega 4096 Jul 19 05:27 Backups
~/Dygma 
🐧 ❯ cd Backups/
~/Dygma/Backups 
🐧 ❯ ls -l
total 16
-rw-rw-r-- 1 lrvega lrvega 14975 Jul 19 20:19 VirtualRaiseANSI.json
```
My user, group, and others have read/write permissions to the entire directory tree for `Dygma`. For the file created with Bazecor 1.3.x, user/group have read/write access and the others has read. These permissions seem correct for the type of operations the program is trying to do (read and write a file under the user's id). But just to test I changed the permissions:
```
 chmod -R 777 Dygma/
🐧 ❯ ls -l | grep Dygma
drwxrwxrwx  3 lrvega lrvega   4096 Jul 13 08:16 Dygma
 cd Dygma/
~/Dygma 
🐧 ❯ ls -l
total 4
drwxrwxrwx 2 lrvega lrvega 4096 Jul 19 05:27 Backups
🐧 ❯ cd Backups/
~/Dygma/Backups 
🐧 ❯ ls -l
total 16
-rwxrwxrwx 1 lrvega lrvega 14975 Jul 19 20:19 VirtualRaiseANSI.json
```
Bazecor 1.4.x still cannot create a virtual keyboard. If you run a 24.04 Ubuntu VM it will 100% repro.

**Comment by noticons** [comment]
@alexpargon I chmodded a directory as a test with 777 and pointed both the backup directory of the Bazecor app and the location for the "virtual keyboard" there, but the exact same problem occurred with the same ancillary issues.

**Comment by noticons** [comment]
@lrvega according to your output, are you indicating that a .json file is actually produced by creating the virtual keyboard? I don't even get a file at all.

**Comment by lrvega** [comment]
@noticons that file was created by BazeCore 1.3.x not by 1.4.x

**Comment by alexpargon** [comment]
@noticons @lrvega  i will be checking this issue as it makes no sense that not even the file is being created 🤪 

let me take a look 👀

**Comment by alexpargon** [comment]
Hey there, so after looking into it, I found a very weird behavior related to this bug: https://github.com/electron/electron/issues/42621

It fails always because the name comes dynamically depending on the file you selected, so sorry but the bug is in Electron right now, as soon as they release a version with that patch, I will merge it with the branch.

**Comment by lrvega** [comment]
@alexpargon thanks for tracking it down! Seems like they merged into main last month https://github.com/electron/electron/pull/42652  Have they not yet produced a release?

**Comment by alexpargon** [comment]
> @alexpargon thanks for tracking it down! Seems like they merged into main last month [electron/electron#42652](https://github.com/electron/electron/pull/42652) Have they not yet produced a release?

They just did! ~14h ago 🤣 

merging..
